runway/module/__init__.py

```diff
@@ -60,67 +24,169 @@ def run_module_command(cmd_list, env_vars):
         sys.exit(shelloutexc.returncode)
 
 
-def use_npm_ci(path):
-    """Return true if npm ci should be used in lieu of npm install."""
-    # https://docs.npmjs.com/cli/ci#description
-    with open(os.devnull, 'w') as fnull:
-        if ((os.path.isfile(os.path.join(path,
-                                         'package-lock.json')) or
-             os.path.isfile(os.path.join(path,
-                                         'npm-shrinkwrap.json'))) and
-                subprocess.call(
-                    [NPM_BIN, 'ci', '-h'],
-                    stdout=fnull,
-                    stderr=subprocess.STDOUT
-                ) == 0):
-            return True
-    return False
-
-
-def run_npm_install(path, options, context):
-    """Run npm install/ci."""
-    # Use npm ci if available (npm v5.7+)
-    if options.get('skip_npm_ci'):
-        LOGGER.info("Skipping npm ci or npm install on %s...",
-                    os.path.basename(path))
-    elif context.env_vars.get('CI') and use_npm_ci(path):  # noqa
-        LOGGER.info("Running npm ci on %s...",
-                    os.path.basename(path))
-        subprocess.check_call([NPM_BIN, 'ci'])
-    else:
-        LOGGER.info("Running npm install on %s...",
-                    os.path.basename(path))
-        subprocess.check_call([NPM_BIN, 'install'])
-
-
-class RunwayModule(object):
+class RunwayModule(object):  # noqa pylint: disable=too-many-instance-attributes
     """Base class for Runway modules."""
 
-    def __init__(self, context, path, options=None):
+    def __init__(self, context, path, runway_file_options=None):
         """Initialize base class."""
+        self.name = os.path.basename(path)
+
         self.context = context
 
+        self.folder = RunwayModuleFolder(path)
+
+        # it would be good to remove the need for sub-classes to refer
+        #  to this directly, and have them rely on 'folder' instead
         self.path = path
 
-        if options is None:
-            self.options = {}
-        else:
-            self.options = options
+        environments_options = better_dict_get(runway_file_options, 'environments', {})
+        self.environment_options = environments_options.get(context.env_name)
+        if self.environment_options is not None:
+            # it will almost always be a 'dict', but some modules (like serverless) support
+            #  a boolean, which indicates we want the module, and we have no values to set
+            if isinstance(self.environment_options, bool):
+                if self.environment_options:
+                    self.environment_options = {}
+                else:
+                    self.environment_options = None
+
+        self.module_options = better_dict_get(runway_file_options, 'options', {})
```

Comment:
I can see why you're looking to simply this now. I had always made the modules grab these so there didn't need to be an explicit name.

I was trying model the module's nested `options` after Serverless Framework's `custom:` key, where any random module-type-specific stuff could go (in contrast with module options themselves, which are the common-across-module-types stuff).

Could call it `self.module_custom_options` to stick with the heritage? Or something totally different?